### DISCUSSION TITLE
Best way to set z-index for menus

### DISCUSSION THREAD

**Discussion by IanVS** [ariakit/ariakit#1697] [Questions] [answered]
Hi, I just got bitten by the menu component not having a z-index, and my menus started bleeding into the rest of my content once I switched over to this project.  I'd like to be able to create my own abstraction around ariakit menus for my project, and ensure that all of my menus render correctly.  But If I try to assign a z-index to that component, if I have two of them close to each other, then the Menu list will not render on top of other menu buttons.  For example:

<img width="293" alt="image" src="https://user-images.githubusercontent.com/4616705/182256029-6d18a373-a083-4660-9c92-c2675ab875a5.png">

How is z-index supposed to be managed when using Menu components from ariakit?

**Comment by IanVS** [comment]
This seems to do the trick, is it the expected way to do this, or a hack?

```ts
const menuState = useMenuState();
if (menuState.contentElement) {
	menuState.contentElement.style.zIndex = '1';
}
```

**Comment by diegohaz** [comment, marked as answer]
You just set `z-index` to the `Menu` element as you would typically do with DOM elements:

```jsx
<Menu style={{ zIndex: 50 }}>
```

or:

```jsx
<Menu className="menu" />
```

```css
.menu {
  z-index: 50;
}
```

Or using any other styling method.

**Reply by IanVS** [reply]
Thanks, I had tried setting z-index with a className as your second example shows, and that does not work.  However, setting the `style` does.  Is it a bug if using css to set the z-index does not work?  Should I file an issue?

**Reply by diegohaz** [reply]
It should work. The examples on the site use class name. An issue with a minimum reproduction would help. 👍

**Reply by IanVS** [reply]
Thanks, I tried it again and it works.  I think what happened before is that it didn't work after a hot-module-reload (js runs to determine what to set the z-index to when it mounts), and I saw it wasn't working.  But after a page refresh, it does.
